### PR TITLE
[fakeintake] redact keys from header at log

### PR DESCRIPTION
**Pull request by pducolin**
### What does this PR do?

Redact header in fakeintake server at log of Datadog payload handling

### Motivation

incident-29228

### CODE REVIEW

File: test/fakeintake/server/server_test.go [hunk 1]
```diff
@@ -654,3 +654,27 @@ func PostSomeRealisticLogs(t *testing.T, url string) {
...
+		"Other-Key": []string{"<redacted>"},
+	}
+	for key, expectedValues := range expectedHeader {
+		redactedValues := redactedHeader[key]
+		assert.Equal(t, expectedValues, redactedValues, "unexpected value at key %s", key)
+	}
+	for key, redactedValues := range redactedHeader {
+		expectedValues := expectedHeader[key]
+		assert.Equal(t, expectedValues, redactedValues, "unexpected value at key %s", key)
+	}
```
Comment (FlorentClarret) [inline test/fakeintake/server/server_test.go:679]: You should also check `expectedHeader` and `redactedValues` lengths are equal I think to be sure we did not miss any headers

**Review by FlorentClarret — APPROVED** [review]
Left two non-blocking comments

File: test/fakeintake/server/server_test.go (hunk 1 above)
Comment (pducolin) [reply test/fakeintake/server/server_test.go:683]: That should be covered by cross-checking keys, if any key is missing in one map it's going to trigger an error at the assertion time, as values is going to be empty.

Added an additional, more explicit check

**Review by FlorentClarret — APPROVED** [review]

File: test/fakeintake/server/server.go [hunk 2]
```diff
@@ -367,6 +367,23 @@ func (fi *Server) handleDatadogRequest(w http.ResponseWriter, req *http.Request)
 	writeHTTPResponse(w, response)
 }
 
+func redactHeader(header http.Header) http.Header {
```
Comment (pgimalac) [inline test/fakeintake/server/server.go:370]: For the record you could also use the agent scrubber, it is defined in its own modules and has no dependencies

**Pull request merged**